src/huggingface_hub/hf_api.py

```diff
@@ -9673,6 +9737,86 @@ def list_user_following(self, username: str, token: Union[bool, str, None] = Non
         ):
             yield User(**followed_user)
 
+    def list_papers(
+        self,
+        *,
+        date: Optional[str] = None,
+        query: Optional[str] = None,
+    ) -> Iterable[PaperInfo]:
+        """
+        List daily papers on the Hugging Face Hub, given a date or a search query.
+
+        Args:
+            date (`str`, *optional*):
+                The date to retrieve papers for, in the format 'YYYY-MM-DD'.
+                If provided, returns papers submitted on this date.
+            query (`str`, *optional*):
+                A search query string to find papers.
+                If provided, returns papers that match the query.
+        Returns:
+            `Iterable[PaperInfo]`: an iterable of [`huggingface_hub.hf_api.PaperInfo`] objects.
+
+        Raises:
+            [`HTTPError`](https://requests.readthedocs.io/en/latest/api/#requests.HTTPError):
+                HTTP 400 if the date is invalid.
+            [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError):
+                If neither `date` nor `query` is provided.
+
+        Example usage with the `date` argument:
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+
+        # List all papers submitted on a specific date
+        >>> api.list_papers(date="2024-09-17")
+        ```
+
+        Example usage with the `query` argument:
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+
+        # List all papers with "attention" in their title
+        >>> api.list_papers(query="attention")
+        ```
+        """
+        if date is None and query is None:
+            raise ValueError("Provide one of `date` or `query`.")
+        path = f"{self.endpoint}/api/daily_papers"
```

Comment:
```suggestion
        path = f"{self.endpoint}/api/papers/search"
```

I would use the `/api/papers/search` API instead as it supports the "query" parameter which `/api/daily_papers` don't. Let's not support `"date"` for this PR so that we have only a single method (`list_papers`) that calls a single route. If in the future we need to be able to search by date, we will update the route server-side to accept it. 

I'm not a fan of having 2 methods for `list_papers` + `list_daily_papers` as they would be confusing. But I'm not a fan either of calling two different routes depending on the search query. That's why I suggest to support only 1 method == 1 route (that currently supports only the `"q"` parameter) and extend later if needed.

Reply:
So what I would suggest is:
- keep `list_papers`
- keep `PaperInfo` as is
- remove `date` argument
- update internal logic to remove `date` + use `/api/papers/search`
- update docstring